internal/fuzzbinder/fuzzclaim.go

```diff
@@ -0,0 +1,111 @@
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package fuzzbinder
+
+// This file provides a custom `ClaimSpec` type, FuzzClaimSpec, to be used
+// for fuzzing claims within the ClaimPredicate (defined in amber package).
+// FuzzClaimSpec is intended to be used for providing the user with the
+// needed elements to characterize the security of a revision of the source
+// code based on fuzzing.
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/project-oak/transparent-release/pkg/amber"
+)
+
+// FuzzClaimV1 is the URI that should be used as the ClaimType in V1 Amber
+// Claim representing a V1 Fuzz Claim.
+const FuzzClaimV1 = "https://github.com/project-oak/transparent-release/fuzz_claim/v1"
+
+// FuzzClaimSpec gives the `ClaimSpec` definition.
+type FuzzClaimSpec struct {
+	// `ClaimSpec` per fuzz-target.
+	PerTarget []PerTargetSpec `json:"perTarget"`
+	// `ClaimSpec` for all the fuzz-targets.
+	PerProject *PerProjectSpec `json:"perProject"`
+}
+
+// PerTargetSpec contains the fuzzing claims specification per fuzz-target.
+type PerTargetSpec struct {
+	// Name of the fuzz-target.
+	Name string `json:"name"`
+	// URI of the fuzz-target.
+	URI string `json:"uri"`
+	// Coverage specifies the code coverage by the fuzz-target.
+	Coverage *FuzzCoverage `json:"coverage"`
+	// Bugs specifies the number of detected bugs using the fuzz-target.
+	Bugs int `json:"bugs"`
+	// Crashes specifies the number of crashes obtained using the fuzz-target.
+	Crashes int `json:"crashes"`
+	// FuzzEffort specifies the fuzzing efforts spent while using the fuzz-target.
+	FuzzEffort *FuzzEffortSpec `json:"fuzzEffort,omitempty"`
+}
+
+// PerProjectSpec contains the fuzzing claims specification of the revision
+// of the source code for all fuzz-targets.
+type PerProjectSpec struct {
+	// Coverage specifies the code coverage by all fuzz-targets.
+	Coverage *FuzzCoverage `json:"coverage"`
+	// Bugs specifies the  number of detected bugs using all fuzz-targets.
+	Bugs int `json:"bugs"`
+	// Crashes specifies the  number of crashes obtained using all fuzz-targets.
+	Crashes int `json:"crashes"`
+	// FuzzEffort specifies the fuzzing efforts spent while using all fuzz-targets.
+	FuzzEffort *FuzzEffortSpec `json:"fuzzEffort,omitempty"`
+}
+
+// FuzzCoverage contains the code coverage by fuzz testing.
+type FuzzCoverage struct {
+	// Line specifies line coverage.
+	Line string `json:"line"`
+	// Branch specifies branch coverage.
+	Branch string `json:"branch"`
+}
+
+// FuzzEffortSpec contains the fuzzing efforts.
+type FuzzEffortSpec struct {
+	// FuzzTime specifies the fuzzing time.
+	FuzzTime *time.Duration `json:"fuzzTime,omitempty"`
```

Comment:
I don't know how `time.Duration` is serialized into JSON. Does the serialization in JSON include the unit (seconds, minutes, etc)?

Reply:
I replaced it with `int` (to use the time in seconds) for simplicity. However, it's possible to use the `time.Duration`  but not directly as described here: https://biscuit.ninja/posts/go-unmarshalling-json-into-time-duration/